tests/integration/test_hyper_transformer.py

```diff
@@ -1,9 +1,100 @@
+from unittest.mock import patch
+
 import numpy as np
 import pandas as pd
 import pytest
 
 from rdt import HyperTransformer
-from rdt.transformers import OneHotEncodingTransformer
+from rdt.transformers import BaseTransformer, OneHotEncodingTransformer, get_default_transformer
+
+
+class DummyTransformer(BaseTransformer):
+
+    INPUT_TYPE = 'any'
+    OUTPUT_TYPES = {
+        'value': 'float'
+    }
+
+    def _fit(self, data):
+        pass
+
+    def _transform(self, data):
+        return data
+
+    def _reverse_transform(self, data):
+        return data
+
+
+class DummyTransformerNumerical(BaseTransformer):
+
+    INPUT_TYPE = 'any'
+    OUTPUT_TYPES = {
+        'value': 'float'
+    }
+
+    def _fit(self, data):
+        pass
+
+    def _transform(self, data):
+        return data.astype(float)
+
+    def _reverse_transform(self, data):
+        return data.astype(str)
+
+
+class DummyTransformerNotMLReady(BaseTransformer):
+
+    INPUT_TYPE = 'datetime'
+    OUTPUT_TYPES = {
+        'value': 'categorical',
+    }
+
+    def _fit(self, data):
```

Comment:
Seeing that a no-op `_fit` is added quite often, maybe we want to actually make the one from the `BaseTransformer` a no-op method instead of an abstract one? I would not do it in this PR, but we can open an issue to do evaluate it and maybe do it sometime in the future